source/Network/Infrastructure/ConnectionServerSM.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Common;
+using NLog;
+using Stateless;
+using System;
+using System.Collections.Generic;
+
+namespace Network.Infrastructure
+{
+    using StateConfiguration = StateMachine<EServerConnectionState, EServerConnectionTrigger>.StateConfiguration;
+    public enum EServerConnectionTrigger
+    {
+        Close,
+        ClientInfoVerified,
+        ClientReady,
+    }
+
+    public enum EServerConnectionState
+    {
+        /// <summary>
+        /// Server is awaiting connection
+        /// </summary>
+        AwaitingClient,
+
+        /// <summary>
+        /// Client is joining.
+        /// </summary>
+        ClientJoining,
+
+        /// <summary>
+        /// Client is ready.
+        /// </summary>
+        Ready,
```

Comment:
Ready for what?

Reply:
"Ready" is kind of a placeholder for a valid state for the CoopServerSM to take control. If you have an idea of a more explicit name let me know.